src/libYARP_OS/src/Time.cpp

```diff
@@ -213,11 +215,15 @@ void Time::useNetworkClock(const ConstString& clock, ConstString localPortName)
 }
 
 void Time::useCustomClock(Clock *clock) {
-    if(clock == nullptr)
+    if(clock == nullptr) {
         YARP_FAIL(Logger::get(), "failed configuring CustomClock client");
+        return;
+    }
```

Comment:
YARP_FAIL calls `exit`, so `return` here is redundant but made the code more readable (and may static analyzer are more happy)
Shall we update all `YARP_FAIL`s? look line 181 and 196

Reply:
I imagines, but as you say:
- it is not clear for people reading the code
- the static analyzer was not happy! So I preferred being explicit (given that this did not change the behaviour)

Reply:
anyway, I added the other two returns